src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -44,6 +205,469 @@ private void BreakPointWithDebuggerFuncEval()
+                /// to efficiently fetch that property from a .NET object (See Fetch method).  
+                /// It hides some slightly complex generic code.  
+                /// </summary>
+                class PropertyFetch
+                {
+                    /// <summary>
+                    /// Create a property fetcher from a .NET Reflection PropertyInfo class that
+                    /// represents a property of a paritcular type.  
+                    /// </summary>
+                    public static PropertyFetch FetcherForProperty(PropertyInfo propertyInfo)
+                    {
+                        if (propertyInfo == null)
+                            return new PropertyFetch();     // returns null on any fetch.
+
+                        var typedPropertyFetcher = typeof(TypedFetchProperty<,>);
+                        var instantiatedTypedPropertyFetcher = typedPropertyFetcher.GetTypeInfo().MakeGenericType(
+                            propertyInfo.DeclaringType, propertyInfo.PropertyType);
+                        return (PropertyFetch)Activator.CreateInstance(instantiatedTypedPropertyFetcher, propertyInfo);
+                    }
+
+                    /// <summary>
+                    /// Given an object, fetch the property that this propertyFech represents. 
+                    /// </summary>
+                    public virtual object Fetch(object obj) { return null; }
+
+                    #region private 
+
+                    class TypedFetchProperty<Tobject, Tproperty> : PropertyFetch
+                    {
+                        public TypedFetchProperty(PropertyInfo property)
+                        {
+                            _propertyFetch = (Func<Tobject, Tproperty>)property.GetMethod.CreateDelegate(typeof(Func<Tobject, Tproperty>));
+                        }
+                        public override object Fetch(object obj)
+                        {
+                            return _propertyFetch((Tobject)obj);
+                        }
+                        private Func<Tobject, Tproperty> _propertyFetch;
+                    }
+                    #endregion
+                }
+
+                string _propertyName;
+                Type _expectedType;
+                PropertyFetch _fetchForExpectedType;
+                #endregion
+            }
+
+            string _outputName;
+            PropertySpec _fetches;
+            #endregion
+        }
+
+        /// <summary>
+        /// CallbackObserver is a adapter class that creats an observer (which you can pass
+        /// to IObservable.Subscribe), and calls the given callback every time the 'next' 
+        /// operation on the IObserver happens. 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        internal class CallbackObserver<T> : IObserver<T>
+        {
+            public CallbackObserver(Action<T> callback) { _callback = callback; }
+
+            #region private 
```

Comment:
This is debatable.    For me, private means 'Clients do not need to (and would prefer not to) see this'  

This includes all privates, but it also includes implementations of contracts that a client is not likely to call directly.   Thus implementations of Dispose() or in this case IObservable all into this category (showing them to clients does not really tell you anything as the public inheritance contract already says it is IDisposable or IObservable, and the fact that there is an explicit implementation is a hidable detail) so I hide it.

The goal is to segregate nicely what a CLIENT wants to see.